src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java

```diff
@@ -25,28 +29,81 @@ public ListCommand parse(String args) throws ParseException {
         if (trimmedArgs.isEmpty()) {
             return new ListCommand();
         }
-
-        // Flags are delimited by whitespace
-        final String[] flags = trimmedArgs.split("\\s+");
-        if (!hasValidFlagCombination(flags)) {
+        // Preamble is necessary due to how ArgumentTokenizer works
+        // TODO modify ArgumentTokenizer to be able to detect without preamble
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize("preamble " + trimmedArgs,
+                                                                       toPrefixArray(ListCommand.VALID_PREFIXES));
+        ArrayList<String> prefixesPresent = getPresentPrefixes(argumentMultimap, ListCommand.VALID_PREFIXES);
+        if (!hasValidPrefixCombination(prefixesPresent)) {
             throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
                                                    ListCommand.MESSAGE_USAGE));
         }
+        Calendar startDate = getStartDate(argumentMultimap);
+        Calendar endDate = getEndDate(argumentMultimap);
+
+        return new ListCommand(startDate, endDate, prefixesPresent);
+    }
+
+    private Calendar getStartDate(ArgumentMultimap argumentMultimap) {
+        ArrayList<String> dateArgs = (ArrayList<String>) argumentMultimap.getAllValues(CliSyntax.PREFIX_FROM);
+        return getDate(getString(dateArgs).trim());
+    }
+
+    private Calendar getEndDate(ArgumentMultimap argumentMultimap) {
+        ArrayList<String> dateArgs = (ArrayList<String>) argumentMultimap.getAllValues(CliSyntax.PREFIX_TO);
+        return getDate(getString(dateArgs).trim());
+    }
+
+    private Calendar getDate(String rawDate) {
+        if (rawDate.isEmpty()) {
+            return null;
+        }
+        GregorianCalendar date = new GregorianCalendar();
+        date.setTime(((new PrettyTimeParser().parse(rawDate)).get(0)));
```

Comment:
catch NullPointerException here. ptp returns an empty list if it fails to parse the arguments. 
alternatively, use the getDate(String rawDate) from my logic.parser.ParserUtil